pkg/render/node_test.go

```diff
@@ -2383,16 +2369,42 @@ func verifyProbes(ds *apps.DaemonSet, isOpenshift, isEnterprise bool) {
 			Port: intstr.FromInt(9099),
 		}}}
 
-	switch {
-	case isOpenshift && isEnterprise:
-		expectedReadiness.Exec.Command = []string{"/bin/calico-node", "-bird-ready", "-bgp-metrics-ready"}
+	if isOpenshift {
 		expectedLiveness.HTTPGet.Port = intstr.FromInt(9199)
-	case isOpenshift && !isEnterprise:
-		expectedReadiness.Exec.Command = []string{"/bin/calico-node", "-bird-ready"}
-		expectedLiveness.HTTPGet.Port = intstr.FromInt(9199)
-	case !isOpenshift && isEnterprise:
+	}
+
+	var found bool
+	var bgp bool
+	for _, env := range ds.Spec.Template.Spec.Containers[0].Env {
+		if env.Name == "CLUSTER_TYPE" {
+			if strings.Contains(env.Value, ",bgp") {
+				bgp = true
+			}
+			found = true
+			break
+		}
+	}
+	Expect(found).To(BeTrue())
+
+	switch {
+	case !isOpenshift && !isEnterprise && !bgp:
```

Comment:
I think isOpenshift can be omitted from these switch cases since it now no longer impacts the Readiness command.